news-list/src/components/NewsItem/NewsItem.spec.js

```diff
@@ -0,0 +1,42 @@
+import { mount } from '@vue/test-utils'
+import NewsItem from '@/components/NewsItem/NewsItem.vue'
+
+describe('NewsItem.vue', () => {
+  describe(':buttons', () => {
+    let newsItem
+    beforeEach(() => {
+      newsItem = mount(NewsItem, {
+        propsData: {
+          item: { id: 20, title: 'Item 1', votes: 0 }
+        }
+      })
+    })
+    describe('click remove', () => {
+      it('emits removeitem event', async () => {
+        const removeButton = newsItem.find('button.remove-button')
+        await removeButton.trigger('click')
+        expect(newsItem.emitted()).toEqual({ removeitem: [[20]] })
+      })
+    })
+
+    describe('click upvote', () => {
+      it('increases votes by one', async () => {
+        const upvoteButton = newsItem.find('button.upvote-button')
+        upvoteButton.trigger('click')
+        await newsItem.vm.$nextTick()
+
+        expect(newsItem.vm._data.mutableItem.votes).toBe(1)
+      })
+    })
+
+    describe('click downvote', () => {
+      it('decreases votes by one', async () => {
+        const upvoteButton = newsItem.find('button.downvote-button')
+        upvoteButton.trigger('click')
+        await newsItem.vm.$nextTick()
+
+        expect(newsItem.vm._data.mutableItem.votes).toBe(-1)
```

Comment:
@roschaefer To avoid mutating the property in the component itself, I added a new data attribute in the NewsItem.vue. In order to get the test case working I had to access the new item "mutableItem" in that way. But I am not really sure if that is the best way to do it.

Reply:
The best idea here is to assert that an event was emitted properly:

1. Arrange: Create a component with `propsData: { item: { id: 1, title: 'Some title', votes :3 } }`
2. Act: `await newsItem.find('button.downvote-button').trigger('click')`
3. Assert `expect(newsItem.emitted('updatedItem')).toEqual([[{ id: 1, title: 'Some title', votes: 2 }]])`

Testing private state like `vm._data` shouldn't be necessary. Always assert something you "see" (like the rendered DOM) or the public interface (props, events).